pkg/integrations/manager_test.go

```diff
@@ -37,6 +37,50 @@ func TestManager_ValidInstanceConfigs(t *testing.T) {
 	})
 }
 
+// TestManager_NoIntegrationsScrape ensures that configs don't get generates
+// when the ScrapeIntegrations flag is disabled.
+func TestManager_NoIntegrationsScrape(t *testing.T) {
+	mock := newMockIntegration()
+
+	integrations := []Integration{mock}
+	im := instance.NewBasicManager(instance.DefaultBasicManagerConfig, log.NewNopLogger(), mockInstanceFactory, func(c *instance.Config) error {
+		globalConfig := prom_config.DefaultConfig.GlobalConfig
+		return c.ApplyDefaults(&globalConfig)
+	})
+
+	cfg := mockManagerConfig()
+	cfg.ScrapeIntegrations = false
+
+	m, err := newManager(cfg, log.NewNopLogger(), im, integrations)
+	require.NoError(t, err)
+	defer m.Stop()
+
+	time.Sleep(time.Second)
+	require.Zero(t, len(im.ListConfigs()))
+}
+
+// TestManager_NoIntegrationScrape ensures that configs don't get generates
+// when the ScrapeIntegration flag is disabled on the integration.
+func TestManager_NoIntegrationScrape(t *testing.T) {
+	mock := newMockIntegration()
+
+	noScrape := false
+	mock.commonCfg.ScrapeIntegration = &noScrape
+
+	integrations := []Integration{mock}
+	im := instance.NewBasicManager(instance.DefaultBasicManagerConfig, log.NewNopLogger(), mockInstanceFactory, func(c *instance.Config) error {
+		globalConfig := prom_config.DefaultConfig.GlobalConfig
+		return c.ApplyDefaults(&globalConfig)
+	})
+
+	m, err := newManager(mockManagerConfig(), log.NewNopLogger(), im, integrations)
+	require.NoError(t, err)
+	defer m.Stop()
+
+	time.Sleep(time.Second)
+	require.Zero(t, len(im.ListConfigs()))
```

Comment:
test.Poll instead of sleep?

Reply:
`test.Poll` wouldn't work here since it's already 0 by default. We need to make sure it's still zero after some period. I'll add a comment to explain this